File: project2/week1/Hyungjun/IntervalListIntersections.go
func intervalIntersection(firstList [][]int, secondList [][]int) [][]int {
	p1, p2 := 0, 0
	var result [][]int
	for p1 < len(firstList) && p2 < len(secondList) {
		if firstList[p1][0] < secondList[p2][0] {
			if firstList[p1][1] < secondList[p2][0] {
				p1 += 1
			} else if firstList[p1][1] <= secondList[p2][1] {
				result = append(result, []int{secondList[p2][0], firstList[p1][1]})
				p1 += 1
			} else {
				result = append(result, []int{secondList[p2][0], secondList[p2][1]})
				p2 += 1
			}
		} else { // fistList[p1][0] >= secondList[p2][0]
			if secondList[p2][1] < firstList[p1][0] {
				p2 += 1
			} else if secondList[p2][1] <= firstList[p1][1] {
				result = append(result, []int{firstList[p1][0], secondList[p2][1]})
				p2 += 1
			} else {
				result = append(result, []int{firstList[p1][0], firstList[p1][1]})
				p1 += 1
			}
		}
	}

	return result
}

File: project2/week1/Sungwon/Interval List Intersections.py
class Solution:
    def intervalIntersection(self, firstList: List[List[int]], secondList: List[List[int]]) -> List[List[int]]:
        first, second = 0, 0
        answer = []
        while first < len(firstList) and second < len(secondList):
            if firstList[first][0] < secondList[second][0]:
                if firstList[first][1] >= secondList[second][0]:
                    if firstList[first][1] <= secondList[second][1]:
                        answer.append([secondList[second][0], firstList[first][1]])
                        first += 1
                    else:
                        answer.append([secondList[second][0], secondList[second][1]])
                        second += 1
                else:
                    first += 1
            else:
                if firstList[first][0] <= secondList[second][1]:
                    if firstList[first][1] <= secondList[second][1]:
                        answer.append([firstList[first][0], firstList[first][1]])
                        first += 1
                    else:
                        answer.append([firstList[first][0], secondList[second][1]])
                        second += 1
                else:
                    second += 1
        return answer
